libraries/chain/block.cpp

```diff
@@ -111,4 +112,22 @@ namespace eosio { namespace chain {
       return validate_and_extract_block_extensions( block_extensions );
    }
 
+   signed_block_ptr pruned_block::to_signed_block() const {
+      if (prune_state != prune_state_type::complete_legacy)
+         return signed_block_ptr{};
+
+      auto result = std::make_shared<signed_block>(*this);
```

Comment:
I would make this `auto result = std::make_shared<signed_block>(static_cast<const signed_block_header&>(*this));` so the slice is obvious.